### PR TITLE
[distribution_list] Improve distribution list by...

### PR DESCRIPTION
**Pull request by JonathanNEMRY**
... fixing `Mass mailing` feature that failed due to a frozen dict (context)
and adding a test to the `mail_compose_message` used with `distribution_list_id`

### CODE REVIEW

File: distribution_list/tests/test_mail_compose_message.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            'email': email_test,
+            'notify_email': 'always',
+        })
+
+        distribution_list_line_id = self.distri_list_line_obj.create(
+            cr, uid,
+            {'name': '%s' % uuid4(),
+             'domain': "[['name', '=', '%s']]" % sample_values,
+             'src_model_id': dst_model_id,
+             })
```
Comment (Olivier-LAURENT) [inline distribution_list/tests/test_mail_compose_message.py]: usually context must be propagated ...

File: distribution_list/tests/test_mail_compose_message.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+        distribution_list_id = self.distri_list_obj.create(
+            cr, uid,
+            {'name': '%s' % uuid4(),
+             'dst_model_id': dst_model_id,
+             'to_include_distribution_list_line_ids':
+                 [[4, distribution_list_line_id]]
+             })
```
Comment (Olivier-LAURENT) [inline distribution_list/tests/test_mail_compose_message.py]: idem

File: distribution_list/tests/test_mail_compose_message.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        dst_model_id = self.registry('ir.model').search(
+            self.cr, self.uid, [('model', '=', 'res.partner')])[0]
+
+        email_test = '%s' % '%s@test.te' % (sample_values)
+
+        partner_id = self.partner_obj.create(cr, uid, {
+            'name': sample_values,
+            'email': email_test,
+            'notify_email': 'always',
+        })
```
Comment (Olivier-LAURENT) [inline distribution_list/tests/test_mail_compose_message.py]: idem

**Comment by Olivier-LAURENT** [comment]
:+1:

**Comment by laetitia-gangloff** [comment]
:+1:

**Pull request merged**